src/com/github/alex_moon/porcupi/manager/ManagerClient.java

```diff
@@ -3,28 +3,49 @@
 import java.io.BufferedReader;
 import java.io.IOException;
 import java.io.InputStreamReader;
-import java.io.PrintWriter;
 import java.net.Socket;
 import java.net.UnknownHostException;
-import java.util.ArrayList;
-import java.util.List;
 
+import com.github.alex_moon.porcupi.messages.Message;
+import com.github.alex_moon.porcupi.messages.MessageException;
 import com.github.alex_moon.porcupi.Config;
 import com.github.alex_moon.porcupi.transport.Transport;
 
 public class ManagerClient implements Manager {
     private String inputLine;
+    private String context;
 
     public ManagerClient() {
         Socket socket;
         Transport transport;
         try {
+            String action, tokens;
             socket = new Socket("127.0.0.1", Config.managerPort);
             BufferedReader stdIn = new BufferedReader(new InputStreamReader(System.in));
             transport = new Transport(socket, this);
             transport.start();
             while ((inputLine = stdIn.readLine()) != null) {
-                transport.tell(inputLine);
+                if (!inputLine.trim().equals("")) {
+                    if (context != null) {
+                        action = context;
+                        tokens = inputLine;
+                    } else {
+                        action = inputLine.split(" ")[0];
```

Comment:
Too much logic in this method - split these out into other methods - indeed, why not a `ManagerClientParser`?